core/parser.mly

```diff
@@ -1281,7 +1281,7 @@ regex_pattern_sequence:
  */
 pattern:
 | typed_pattern                                             { $1 }
-| typed_pattern COLON primary_datatype                      { (`HasType ($1, datatype $3), pos()) }
+| typed_pattern COLON primary_datatype                      { (`HasType ($1, datatype ($3, pos())), pos()) }
```

Comment:
This new usage of `pos()` is incorrect - it annotates `$3` with position of the whole production, rather than just with position of `$3`. Fixing it would require pushing `pos()` into datatype production. Since this would require a lot of clutter I decided to avoid it for now - but see https://github.com/links-lang/links/issues/411#issuecomment-438246200